setup/blobfuse2.service

```diff
@@ -47,7 +48,7 @@ Environment=attr_timeout=240
 Environment=entry_timeout=240
 Environment=negative_timeout=120
 Type=simple
-ExecStart=/usr/local/bin/blobfuse2 mount ${BlobMountingPoint} --tmp-path=${BlobTmp} --config-file=${BlobConfigFile} --log-level=${BlobLogLevel}  
+ExecStart=/usr/local/bin/blobfuse2 mount ${BlobMountingPoint} --foreground --tmp-path=${BlobTmp} --config-file=${BlobConfigFile} --log-level=${BlobLogLevel}  
```

Comment:
--foreground will not daemonize the blobfuse binary.

Reply:
When it backgrounds the blobfuse2 process it doesn't stay running.

If I start the service with the following ExecStart it immediately exits.

```
ExecStart=/usr/bin/blobfuse2 mount ${BlobMountingPoint} --tmp-path=${BlobTmp} --config-file=${BlobConfigFile} --log-level=${BlobLogLevel}
```

```
sudo systemctl start blobfuse2.service
sudo systemctl status blobfuse2.service
blobfuse2.service - A virtual file system adapter for Azure Blob storage.
     Loaded: loaded (/etc/systemd/system/blobfuse2.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Fri 2022-12-09 14:54:10 AEDT; 478ms ago
    Process: 8742 ExecStart=/usr/bin/blobfuse2 mount ${BlobMountingPoint} --tmp-path=${BlobTmp} --config-file=${BlobCon>
    Process: 8752 ExecStop=/usr/bin/fusermount -u ${BlobMountingPoint} (code=exited, status=1/FAILURE)
   Main PID: 8742 (code=exited, status=0/SUCCESS)
        CPU: 227ms

Dec 09 14:54:08 tgn-rd-bk01 systemd[1]: Started A virtual file system adapter for Azure Blob storage..
Dec 09 14:54:09 tgn-rd-bk01 blobfuse2[8742]: LOG_CRIT [mount.go (364)]: Starting Blobfuse2 Mount : 2.0.1 on [Ubuntu >
Dec 09 14:54:09 tgn-rd-bk01 blobfuse2[8742]: LOG_CRIT [mount.go (365)]: Logging level set to : LOG_WARNING
Dec 09 14:54:10 tgn-rd-bk01 fusermount[8752]: /usr/bin/fusermount: entry for /u1/azure not found in /etc/mtab
Dec 09 14:54:10 tgn-rd-bk01 systemd[1]: blobfuse2.service: Control process exited, code=exited, status=1/FAILURE
Dec 09 14:54:10 tgn-rd-bk01 systemd[1]: blobfuse2.service: Failed with result 'exit-code'.
```

If I start it with `--foreground` it will stay up and running:

```
ExecStart=/usr/bin/blobfuse2 mount ${BlobMountingPoint} --foreground --tmp-path=${BlobTmp} --config-file=${BlobConfigFile} --log-level=${BlobLogLevel}
```

```
sudo systemctl start blobfuse2.service
sudo systemctl status blobfuse2.service
● blobfuse2.service - A virtual file system adapter for Azure Blob storage.
     Loaded: loaded (/etc/systemd/system/blobfuse2.service; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2022-12-09 14:59:38 AEDT; 4s ago
   Main PID: 8814 (blobfuse2)
      Tasks: 9 (limit: 1030)
     Memory: 9.1M
        CPU: 396ms
     CGroup: /system.slice/blobfuse2.service
             └─8814 /usr/bin/blobfuse2 mount /u1/azure --foreground --tmp-path=/u1/blobfuse2/cache --config-file=/u1/bl>

Dec 09 14:59:38 tgn-rd-bk01 systemd[1]: Started A virtual file system adapter for Azure Blob storage..
Dec 09 14:59:39 tgn-rd-bk01 blobfuse2[8814]: LOG_CRIT [mount.go (364)]: Starting Blobfuse2 Mount : 2.0.1 on [Ubuntu >
Dec 09 14:59:39 tgn-rd-bk01 blobfuse2[8814]: LOG_CRIT [mount.go (365)]: Logging level set to : LOG_WARNING
```

Reply:
I find if I remove the `--foreground` flag and add `Type=forking` instead of `Type=simple` it stays running.

```
[Unit]
Description=A virtual file system adapter for Azure Blob storage.
After=network-online.target
Requires=network-online.target
WorkingDirectory=/u1/blobfuse2
[Service]
User=ja
Group=ja
Environment=BlobMountingPoint=/u1/azure
Environment=BlobTmp=/u1/blobfuse2/cache
Environment=BlobConfigFile=/u1/blobfuse2/blobfuse2.yaml
Environment=BlobLogLevel=LOG_WARNING
Environment=attr_timeout=240
Environment=entry_timeout=240
Environment=negative_timeout=120
Type=forking
ExecStart=/usr/bin/blobfuse2 mount ${BlobMountingPoint} --tmp-path=${BlobTmp} --config-file=${BlobConfigFile} --log-level=${BlobLogLevel}
ExecStop=/usr/bin/fusermount -u ${BlobMountingPoint}
[Install]
WantedBy=multi-user.target
```

Reply:
But adding foreground option also means the shell where you start the service will stay stuck untill you unmount from other shell.